frontend/src/routes/RequestReview.tsx

```diff
@@ -657,6 +667,80 @@ function DatasourceRequestBox({
       : []),
   ];
 
+  /* eslint-disable @typescript-eslint/no-unsafe-call */
+  const fileHandler = async (connectionId: string) => {
+    try {
+      // Create a handle for the file the user wants to save
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const fileHandle = await (window as any).showSaveFilePicker({
+        suggestedName: `${connectionId}.sql`,
+        types: [
+          {
+            description: "SQL Files",
+            accept: {
+              "text/sql": [".sql"],
+            },
+          },
+        ],
+      });
+      return fileHandle;
+    } catch (error) {
+      console.error("Error getting file handle:", error);
+      throw error;
+    }
+  };
+
+  const handleStreamSQLDump = async (connectionId: string) => {
+    try {
+      // Get file handle using the utility function
+      const fileHandle = await fileHandler(connectionId);
+
+      // Fetch and handle SQL dump data
+      const combinedSQL = await getSQLDumpStreamedRequest(connectionId);
+
+      // Create a writable stream that writes the blob to the file
+      const writableStream = await fileHandle.createWritable();
+
+      // Write the combined SQL content to the file
+      await writableStream.write(
+        new Blob([combinedSQL], { type: "text/plain" }),
+      );
+
+      // Close the file and release the stream
+      await writableStream.close();
```

Comment:
here we actually load all the SQL into the browser memory now before writing it to the dump file, I think streaming this directly to the downloaded file is more sensible to not accidentally run into an overflow situation. DB dumps can be very large.

Reply:
Updated to stream the SQL data to a downloaded file!